modules/launcher/src/main/java/org/wso2/broker/Main.java

```diff
@@ -62,13 +61,17 @@ public static void main(String[] args) throws Exception {
             StartupContext startupContext = new StartupContext();
 
             initConfigProvider(startupContext);
-            loadUsers();
             BrokerConfigProvider service = startupContext.getService(BrokerConfigProvider.class);
             BrokerConfiguration brokerConfiguration =
                     service.getConfigurationObject(BrokerConfiguration.NAMESPACE, BrokerConfiguration.class);
             DataSource dataSource = getDataSource(brokerConfiguration.getDataSource());
             startupContext.registerService(DataSource.class, dataSource);
-
+            // Initializing broker auth module
+            UserStoreManager userStoreManager = new UserStoreManagerImpl();
+            BrokerAuthConfiguration securityConfiguration = service
+                    .getConfigurationObject(BrokerAuthConfiguration.NAMESPACE, BrokerAuthConfiguration.class);
+            startupContext.registerService(AuthManager.class, new AuthManager(securityConfiguration, dataSource,
```

Comment:
We followed a pattern where the service registration is done in the specific module we just pass the startup context to it. I think we should also use a service interface with start and stop methods for all these components. Can we do the same for this as well?

Reply:
We should not initialize the UserStoreManagerImpl() if we are using the different auth module.